### ISSUE TITLE
Better fallback for NodeLabel

### ISSUE THREAD

**Issue by Beytoven** [GoogleChrome/lighthouse#10421]
In the case where an element has no innerText, alt, or aria-label values, the getNodeLabel function uses the element's tag name as a fallback. This results in situations like the screenshot below. Perhaps there's a better option than just using tag name because it looks weird.

![image](https://user-images.githubusercontent.com/27858957/75829980-78061a00-5d64-11ea-913e-96ce15c7e351.png)

**Comment by connorjclark** [comment]
I could see using context specific attributes. For anchor, we could fallback to the href attribute.

**Comment by Beytoven** [comment]
> I could see using context specific attributes. For anchor, we could fallback to the href attribute.

I think that makes sense. For anchors, they're pretty much all link tags so href makes sense.

I've been taking a look at the Accessibility audits and it seems like even with elements that have a NodeLabel, it may not even be used as I had thought. For instance, this node blob 
```json
"node": {
              "type": "node",
              "selector": ".text",
              "path": "1,HTML,1,BODY,11,DIV,2,DIV,1,DIV,0,A,1,SPAN",
              "snippet": "<span class=\"text\">About This Author</span>",
              "explanation": "Fix any of the following:\n  Element has insufficient color contrast of 3.33 (foreground color: #ffffff, background color: #f0595c, font size: 12.0pt, font weight: bold). Expected contrast ratio of 4.5:1",
              "nodeLabel": "ABOUT THIS AUTHOR"
            }
```

results in
 
![image](https://user-images.githubusercontent.com/27858957/75834661-8575d100-5d71-11ea-80dc-667541bf22c0.png)

**Comment by paulirish** [comment]
I think there's an extra thing going on here.  For any `type:'node'` items, when we're in devtools, we upgrade them to a [linkified node](https://cs.chromium.org/chromium/src/third_party/devtools-frontend/src/front_end/lighthouse/LighthouseReportRenderer.js?l=41&rcl=393efd6d94e482796e0819bdf242331a37cc320c). visually, these are super short and terse. Also you can see in your last screenshot that our data doesnt even have the `span.text` string.. that's coming from DevTools.

but outside of DevTools, they use an alternate presentation (which is the `snippet` property plus the `nodeLabel`, right?)

---

so we want our presentation of nodes both 1) within devtools and 2) outside of devtools to be good.   

I think you're kinda suggesting that even in the devtools upgrade we should keep the nodeLabel instead of wiping it out.  That seems cool to me. 👍

**Comment by paulirish** [comment]
resolutions.

1. lets keep nodelabel in the devtools case
2. lets later augment with element screenshots https://github.com/GoogleChrome/lighthouse/pull/8797

**Comment by patrickhulce** [comment]
already discussed and resolution reached, p1.5 because we want element screenshots ASAP

**Comment by tannerdolby** [comment]
I recon this can be closed now with #12727 landing @patrickhulce
